### PR TITLE
Object model update

### PR DESCRIPTION
**Pull request by lvdd-rbl**
**Finished first version of the object model : minimum solution where each machine is to be delivered when possible and installed the day following the delivery. So no penaltys associated to the machines and no issue about technicians' rest days to deal with.**

- Can compute costs from a single itinerary (technician or vehicle) to the entire planning horizon : cost of using a technician or a vehicle, costs for each day, distance per unit costs, ... For now, penaltys associated to the machines are not taken into account (minimum solution)
- Finished testing technician itineraries (distances, number of demands, distances, the day following the delivery, etc)
- Added relation between demands and the planning (added a hashmap attribute to track whether the demand is to be supplied or installed)
- Fixed issue with nbDays in Test1 ("2 instead of 3")
- Updated relation between Demand and Itineraries => now a demand is linked to a vehicle itinerary AND a technician itinerary
- Removed position attribute in Customer
- Added method in Point that computes the Euclidian distance between two points (integer portion "by excess")
- UML updated in Trello

**TODO** : 
- [ ] Start coding Test2 with text files
- [ ] Technicians' rest days (can't work for more than 5 days, and if they work 5 days straight, they need to rest 2 days, else 1 day)
- [ ] Penaltys for each supplied machine when they are not installed (for each day)
- [ ] Design patterns (DAO, Singleton, Factory)

### CODE REVIEW

File: src/modele/Technician.java [hunk 1]
```diff
@@ -154,4 +184,40 @@ public boolean addPotentialInstallation(Machine machine) {
...
+     * @todo : to review
+     * @param demand
+     * @param dayNumber
+     * @return true if success
+     */
+    public boolean canInstallDemand(Demand demand, int dayNumber) {
+        boolean canInstall = false;
+        for (Installation installation : this.potentialInstallations) {
+            if (installation.getMachine().equals(demand.getMachine())) {
+                canInstall = true;
```
Comment (Osgilia) [inline src/modele/Technician.java:202]: if (!installation.getMachine().equals(demand.getMachine())) {
                return false

?

File: src/modele/Technician.java [hunk 2]
```diff
@@ -154,4 +184,40 @@ public boolean addPotentialInstallation(Machine machine) {
...
+                    lastDay = currentDayNumber;
+                    nbStraightWorkingDays++;
+                    if (nbStraightWorkingDays > 4) {
+                        return false;
+                    }
+                }
+            }
+        }
+
+        return canInstall;
```
Comment (Osgilia) [inline src/modele/Technician.java:221]: return true

**Review by Osgilia — APPROVED** [review]
merci pour ce moment

**Review by LouisCalypso — APPROVED** [review]

File: src/modele/Technician.java (hunk 1 above)
Comment (lvdd-rbl) [reply src/modele/Technician.java:202]: to check whether the technician can install the machine
I agree Installation should be called Accreditation and potentialInstallations accreditations

File: src/modele/Technician.java (hunk 2 above)
Comment (lvdd-rbl) [reply src/modele/Technician.java:221]: yes

**Pull request merged**